lib/rspec/matchers/expecteds_for_multiple_diffs.rb

```diff
@@ -62,11 +62,11 @@ def self.truncated(description)
       end
 
       def diffs(differ, actual)
-        @expected_list.map do |(expected, diff_label)|
+        @expected_list.inject("") do |output, (expected, diff_label)|
           diff = differ.diff(actual, expected)
-          next if diff.strip.empty?
-          "#{diff_label}#{diff}"
-        end.compact.join("\n")
+          next output if /\A[[:space:]]*\z/ === diff
+          output << diff_label << diff << "\n".freeze
+        end.tap(&:rstrip!)
```

Comment:
The other consistent behaviour would be to allow newlines at the end of a diff (obtainable by removing the final `tap(rstrip!)`). Then these tests would pass, but the following would fail:

```
rspec ./spec/rspec/expectations/fail_with_spec.rb:11 # RSpec::Expectations#fail_with includes a diff if expected and actual are diffable
rspec ./spec/rspec/expectations/fail_with_spec.rb:64 # RSpec::Expectations#fail_with with --color tells the differ to use color
rspec ./spec/rspec/expectations/fail_with_spec.rb:41 # RSpec::Expectations#fail_with with matchers uses matcher descriptions in place of matchers in diffs
rspec ./spec/rspec/matchers/expecteds_for_multiple_diffs_spec.rb:101 # RSpec::Matchers::ExpectedsForMultipleDiffs#message_with_diff returns message with diff and truncated matcher description if it is too long
rspec ./spec/rspec/matchers/expecteds_for_multiple_diffs_spec.rb:81 # RSpec::Matchers::ExpectedsForMultipleDiffs#message_with_diff returns regular message with diff when single expected
rspec ./spec/rspec/matchers/expecteds_for_multiple_diffs_spec.rb:90 # RSpec::Matchers::ExpectedsForMultipleDiffs#message_with_diff returns message with diff and matcher description when single expected with matcher
rspec ./spec/rspec/matchers/expecteds_for_multiple_diffs_spec.rb:50 # RSpec::Matchers::ExpectedsForMultipleDiffs.for_many_matchers has a diff for all matchers with their description
```

This shows that rspec-expectations' own test suite currently expects inconsistent behaviour (likely to match an implementation artifact) wrt newlines at the end of diffs.